packages/contracts-bedrock/scripts/checks/hotfuzz/main.go

```diff
@@ -0,0 +1,161 @@
+}
+
+type TestConfig struct {
+	maxSeconds  int
+	maxRuns     int
+	matchPaths  string
+	currentRuns int
+}
+
+func main() {
+	config := parseFlags()
+
+	tests, err := getTestNames(config.matchPaths)
+	if err != nil {
+		log.Fatalf("Error getting test names: %v", err)
+	}
+
+	if len(tests) == 0 {
+		log.Println("No tests found matching the criteria")
+		return
+	}
+
+	var wg sync.WaitGroup
+	results := make(chan string, len(tests))
+
+	for _, test := range tests {
+		wg.Add(1)
+		go runTest(test, config, &wg, results)
+	}
+
+	go func() {
+		wg.Wait()
+		close(results)
+	}()
+
+	for result := range results {
+		fmt.Println(result)
+	}
+}
+
+func parseFlags() TestConfig {
+	maxSeconds := flag.Int("max-seconds", 120, "maximum seconds per test")
+	maxRuns := flag.Int("max-runs", 1000, "maximum number of fuzz runs per test")
+	matchPaths := flag.String("match-path", "", "path pattern to match for test files")
+	flag.Parse()
+
+	return TestConfig{
+		maxSeconds:  *maxSeconds,
+		maxRuns:     *maxRuns,
+		matchPaths:  *matchPaths,
+		currentRuns: 0,
+	}
+}
+
+func getTestNames(matchPaths string) ([]TestInfo, error) {
+	args := []string{"test", "--fuzz-runs", "1"}
+	if matchPaths != "" {
+		args = append(args, "--match-path", matchPaths)
+	}
+
+	cmd := exec.Command("forge", args...)
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, fmt.Errorf("error creating stdout pipe: %w", err)
+	}
+
+	if err := cmd.Start(); err != nil {
+		return nil, fmt.Errorf("error starting command: %w", err)
+	}
+
+	var tests []TestInfo
+	scanner := bufio.NewScanner(stdout)
+	testPattern := regexp.MustCompile(`\[PASS\] (test[^\s]+)`)
+	contractPattern := regexp.MustCompile(`Ran \d+ tests? for ([^\s]+)`)
+	currentContract := ""
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if matches := contractPattern.FindStringSubmatch(line); len(matches) > 1 {
+			currentContract = matches[1]
+			continue
+		}
+
+		if matches := testPattern.FindStringSubmatch(line); len(matches) > 1 {
+			testName := matches[1]
+			if idx := strings.Index(testName, "("); idx != -1 {
+				testName = testName[:idx]
+			}
+			tests = append(tests, TestInfo{
+				Name:     testName,
+				Contract: currentContract,
+			})
+		}
+	}
+
+	if err := cmd.Wait(); err != nil {
+		return nil, fmt.Errorf("error waiting for command: %w", err)
+	}
+
+	return tests, nil
+}
+
+func runTest(test TestInfo, config TestConfig, wg *sync.WaitGroup, results chan<- string) {
+	defer wg.Done()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(config.maxSeconds)*time.Second)
+	defer cancel()
+
+	args := []string{
+		"test",
+		"--match-test", test.Name,
+		"--match-contract", test.Contract,
+		"--fuzz-runs", strconv.Itoa(config.maxRuns),
+	}
+	if config.matchPaths != "" {
+		args = append(args, "--match-path", config.matchPaths)
+	}
+
+	cmd := exec.CommandContext(ctx, "forge", args...)
+	output, err := cmd.CombinedOutput()
+
+	runsPattern := regexp.MustCompile(`runs: (\d+)`)
```

Comment:
Right now this always returns 0 if it times out, we should use `--show-progress`